pkg/reconciler/taskrun/validate_taskrun_test.go

```diff
@@ -104,26 +105,37 @@ func TestValidateResolvedTask_ValidParams(t *testing.T) {
 			"key3": "val3",
 		}),
 	}}
-	m := []v1beta1.Param{{
-		Name:  "zoo",
-		Value: *v1beta1.NewStructuredValues("a", "b", "c"),
-	}}
-	if err := ValidateResolvedTask(ctx, p, &v1beta1.Matrix{Params: m}, rtr); err != nil {
+	m := &v1beta1.Matrix{
+		Params: []v1beta1.Param{{
+			Name:  "zoo",
+			Value: *v1beta1.NewStructuredValues("a", "b", "c"),
+		}},
+		Include: []v1beta1.IncludeParams{{
+			Name: "build-1",
+			Params: v1beta1.Params{{
+				Name: "include", Value: v1beta1.ParamValue{Type: v1beta1.ParamTypeString, StringVal: "string-1"},
+			}},
+		}},
+	}
+	if err := ValidateResolvedTask(ctx, p, m, rtr); err != nil {
 		t.Fatalf("Did not expect to see error when validating TaskRun with correct params but saw %v", err)
 	}
-
 	t.Run("alpha-extra-params", func(t *testing.T) {
 		ctx := config.ToContext(ctx, &config.Config{FeatureFlags: &config.FeatureFlags{EnableAPIFields: "alpha"}})
-		extra := v1beta1.Param{
+		extraStringParam := v1beta1.Param{
```

Comment:
This test case is trying to test a lot of different things at once. Can we refactor to test one thing at a time? I think it would be good to at least move the "extra params" test into a separate test case or test class. (Also noticing we don't have coverage for "extra object param")

Reply:
Created another test for testing extra params with table-driven testing. Code coverage is 100% now 🎉

Reply:
maybe I'm missing something but I'm not seeing what happened to the tests for invalid params?